pkg/virt-controller/services/template_test.go

```diff
@@ -428,6 +428,52 @@ var _ = Describe("Template", func() {
 				Expect(pod.Spec.ImagePullSecrets[1].Name).To(Equal("pull-secret-1"))
 			})
 		})
+
+		Context("with slirp interface", func() {
+			It("Should have empty port list in the pod manifest", func() {
+				slirpInterface := v1.InterfaceSlirp{}
+				domain := v1.DomainSpec{}
+				domain.Devices.Interfaces = []v1.Interface{{Name: "testnet", InterfaceBindingMethod: v1.InterfaceBindingMethod{Slirp: &slirpInterface}}}
+				vmi := v1.VirtualMachineInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testvmi", Namespace: "default", UID: "1234",
+					},
+					Spec: v1.VirtualMachineInstanceSpec{Domain: domain},
+				}
+
+				pod, err := svc.RenderLaunchManifest(&vmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(len(pod.Spec.Containers)).To(Equal(1))
+				Expect(len(pod.Spec.Containers[0].Ports)).To(Equal(0))
+			})
+			It("Should create a port list in the pod manifest", func() {
+				slirpInterface := v1.InterfaceSlirp{Ports: []v1.Port{{Port: 80}, {Protocol: "UDP", Port: 80}, {Port: 90}}}
+				domain := v1.DomainSpec{}
+				domain.Devices.Interfaces = []v1.Interface{{Name: "testnet", InterfaceBindingMethod: v1.InterfaceBindingMethod{Slirp: &slirpInterface}}}
+				vmi := v1.VirtualMachineInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testvmi", Namespace: "default", UID: "1234",
+					},
+					Spec: v1.VirtualMachineInstanceSpec{Domain: domain},
+				}
+
+				pod, err := svc.RenderLaunchManifest(&vmi)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(len(pod.Spec.Containers)).To(Equal(1))
+				Expect(len(pod.Spec.Containers[0].Ports)).To(Equal(3))
+				Expect(pod.Spec.Containers[0].Ports[0].Name).To(Equal("tcp-80"))
```

Comment:
Ideally, we should check that the array *contains* the expected ports but regardless of their order. But I guess it's harder to implement the check then.